wasm/jit/jit_arm64_test.go

```diff
@@ -184,3 +213,309 @@ func TestArm64Compiler_compileConsts(t *testing.T) {
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			env := newJITEnvironment()
+
+			// Build code.
+			compiler := env.requireNewCompiler(t)
+			err := compiler.emitPreamble()
+			require.NoError(t, err)
+
+			// Setup the location stack so that we push the const on the specified height.
+			compiler.locationStack.sp = tc.stackPointer
+			compiler.locationStack.stack = make([]*valueLocation, tc.stackPointer)
+
+			// Record that that top value is on top.
+			require.Len(t, compiler.locationStack.usedRegisters, 0)
+			loc := compiler.locationStack.pushValueOnStack()
+			if tc.isFloat {
+				loc.setRegisterType(generalPurposeRegisterTypeFloat)
+			} else {
+				loc.setRegisterType(generalPurposeRegisterTypeInt)
+			}
+			// At this point the value must be recorded as being on stack.
+			require.True(t, loc.onStack())
+
+			// Release the stack-allocated value to register.
+			compiler.loadValueOnStackToRegister(loc)
+			require.Len(t, compiler.locationStack.usedRegisters, 1)
+			require.True(t, loc.onRegister())
+
+			// To verify the behavior, increment the value on the register.
+			if tc.isFloat {
+				// For float, we cannot add consts, so load the constant first.
+				err = compiler.emitFloatConstant(false, math.Float64bits(1))
+				require.NoError(t, err)
+				// Then, do the increment.
+				compiler.applyRegisterToRegisterInstruction(arm64.AFADDD, compiler.locationStack.peek().register, loc.register)
+				// Delete the loaded const.
+				compiler.locationStack.pop()
+			} else {
+				compiler.applyConstToRegisterInstruction(arm64.AADD, 1, loc.register)
+			}
+
+			// Release the value to the memory stack so that we can see the value after exiting.
+			compiler.releaseRegisterToStack(loc)
+			compiler.returnFunction()
+
+			// Generate the code under test.
+			code, _, _, err := compiler.compile()
+			require.NoError(t, err)
+
+			// Run native code after growing the value stack, and place the original value.
+			env.engine().builtinFunctionGrowValueStack(tc.stackPointer)
+			env.stack()[tc.stackPointer] = val
+			env.exec(code)
+
+			// JIT status on engine must be returned and stack pointer must end up the specified one.
+			require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
+			require.Equal(t, tc.stackPointer+1, env.stackPointer())
+
+			if tc.isFloat {
+				require.Equal(t, math.Float64frombits(val)+1, env.stackTopAsFloat64())
+			} else {
+				require.Equal(t, uint64(val)+1, env.stackTopAsUint64())
+			}
+		})
+	}
+}
+
+func TestArm64Compiler_compile_Add_Sub_Mul(t *testing.T) {
+	for _, kind := range []wazeroir.OperationKind{
+		wazeroir.OperationKindAdd,
+		wazeroir.OperationKindSub,
+		wazeroir.OperationKindMul,
+	} {
+		kind := kind
+		t.Run(kind.String(), func(t *testing.T) {
+			for _, unsignedType := range []wazeroir.UnsignedType{
+				wazeroir.UnsignedTypeI32,
+				wazeroir.UnsignedTypeI64,
+				wazeroir.UnsignedTypeF32,
+				wazeroir.UnsignedTypeF64,
```

Comment:
Just curious, why those floats are called `Unsigned` while they're actually signed?

Reply:
UnsignedType = UnsignedInteger || Floats = (I32, I64, F32, F64)!

Reply:
I think I will rework around this #210 !